mcl_ip:
  ros__parameters:
    # leishen16
    laser:
        num_vertical_scans: 16
        num_horizontal_scans: 2000
        ground_scan_index: 7
        vertical_angle_bottom: -15.0          # degrees
        vertical_angle_top: 15.0            # degrees
        sensor_mount_angle: 0.0               # degrees
        scan_period: 0.1                    # seconds

    imageProjection:
        segment_valid_point_num: 5
        segment_valid_line_num: 2
        segment_theta: 60.0                      # decrese this value may improve accuracy
        maximum_detection_range: 50.0            # point cloud within this range will be used to project image

mcl_fa:
  ros__parameters:
    laser:
        num_vertical_scans: 16
        num_horizontal_scans: 2000
        ground_scan_index: 7
        vertical_angle_bottom: -15.0          # degrees
        vertical_angle_top: 15.0            # degrees
        sensor_mount_angle: 0.0               # degrees
        scan_period: 0.1                    # seconds

    featureAssociation:
        odom_type: "wheel_odometry"         #wheel_odometry/laser_odometry
        robot_frame: "base_link"
        sensor_frame: "laser_link"
        sensor_x: 0.3
        sensor_y: 0.0
        sensor_z: 0.38
        sensor_roll: 0.0
        sensor_pitch: 0.0
        sensor_yaw: -3.1415926
        edge_threshold: 0.1
        surf_threshold: 0.1
        nearest_feature_search_distance: 3.0

